src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -245,6 +257,9 @@ private void WatchForFileSystemEventsThreadStart(object arg)
 
                         // When we get here, we've requested to stop so cleanup the EventStream and unschedule from the RunLoop
                         Interop.EventStream.FSEventStreamStop(_eventStream);
+
+                        // Release the lock on the root directory
+                        UnlockRootDirectory(_rootHandle);
```

Comment:
I was relying on the underlying SafeFileHandle to unlock in this case, but I'll make it explicit and add a Catch statement and unlock in there